app/views/contests/scoreboard.html.erb

```diff
@@ -29,20 +39,25 @@
             (<%= @weighting[problem.id] %>)
           </th>
         <% end %>
-        <th style="text-align: right"> Score </th>
-        <th> Time </th>
-        <th> Rank </th>
+        <th style="width:1%; text-align: right"> Score </th>
+        <th style="width:1%"> Time </th>
       </tr>
     </thead>
     <tbody>
-      <% median = @scoreboard[@scoreboard.length/2-1] %>
       <% rank = 1 %>
+      <% num_unofficial = 0 %>
       <% previous_record = @scoreboard.first %>
       <% @scoreboard.each_with_index do |record,index| %>
-        <% if !user_signed_in? or record.user && record.user.id != current_user.id && !policy(@contest).inspect? %>
-          <% next if record.score < median.score || (record.score == median.score && record.time_taken > median.time_taken) # no permission to view %>
-        <% end %>
-        <tr <% if user_signed_in? && record.user && record.user.id == current_user.id %> class="emphasized"<% end %>>
+        <% next if !show_scoreboard && (!user_signed_in? or record.user && record.user.id != current_user.id) # no permission to view %>
```

Comment:
This inherits the bug fixed by #117 (deleted users will be shown to signed-in users when the scoreboard is not live). I'd also like to refactor as in #118, i.e.
```suggestion
        <% next unless show_scoreboard || (user_signed_in? && record.user && record.user.id == current_user.id) # hide row unless permitted to view %>
```
(Might be worth saving `user_signed_in? && record.user && record.user.id == current_user.id` in a variable, because it's also used below.)